src/femr/labelers/omop.py

```diff
@@ -1,49 +1,206 @@
+    "Pleural Other",
+    "Fracture",
+    "Support Devices",
+]
 
 
 def identity(x: Any) -> Any:
     return x
 
 
 def get_visit_concepts() -> List[str]:
-    return ["Visit/IP"]
+    return ["Visit/IP", "Visit/OP"]
 
 
 def get_inpatient_admission_concepts() -> List[str]:
     return ["Visit/IP"]
 
 
+def get_outpatient_visit_concepts() -> List[str]:
+    return ["Visit/OP"]
+
+
 def get_death_concepts() -> List[str]:
     return [
-        "Death Type/OMOP generated",
         "Condition Type/OMOP4822053",
     ]
 
 
+def get_icu_visit_detail_concepts() -> List[str]:
+    return [
+        # All care sites with "ICU" (case insensitive) in the name
+        "CARE_SITE/7928450",
+        "CARE_SITE/7930385",
+        "CARE_SITE/7930600",
+        "CARE_SITE/7928852",
+        "CARE_SITE/7928619",
+        "CARE_SITE/7929727",
+        "CARE_SITE/7928675",
+        "CARE_SITE/7930225",
+        "CARE_SITE/7928759",
+        "CARE_SITE/7928227",
+        "CARE_SITE/7928810",
+        "CARE_SITE/7929179",
+        "CARE_SITE/7928650",
+        "CARE_SITE/7929351",
+        "CARE_SITE/7928457",
+        "CARE_SITE/7928195",
+        "CARE_SITE/7930681",
+        "CARE_SITE/7930670",
+        "CARE_SITE/7930176",
+        "CARE_SITE/7931420",
+        "CARE_SITE/7929149",
+        "CARE_SITE/7930857",
+        "CARE_SITE/7931186",
+        "CARE_SITE/7930934",
+        "CARE_SITE/7930924",
+    ]
+
+
 def move_datetime_to_end_of_day(date: datetime.datetime) -> datetime.datetime:
-    return date.replace(hour=23, minute=59, second=0)
+    return date.replace(hour=23, minute=59, second=59)
 
 
-def get_visit_codes(ontology: extension_datasets.Ontology) -> Set[str]:
-    return {y for x in get_visit_concepts() for y in _get_all_children(ontology, x)}
+def does_exist_event_within_time_range(
+    patient: Patient, start: datetime.datetime, end: datetime.datetime, exclude_event_idxs: List[int] = []
+) -> bool:
+    """Return True if there is at least one event within the given time range for this patient.
+    If `exclude_event_idxs` is provided, exclude events with those indexes in `patient.events` from the search."""
+    excluded = set(exclude_event_idxs)
+    for idx, e in enumerate(patient.events):
+        if idx in excluded:
+            continue
+        if start <= e.start <= end:
+            return True
+    return False
 
 
-def get_inpatient_admission_codes(
+def get_femr_codes(
     ontology: extension_datasets.Ontology,
+    omop_concept_codes: List[str],
+    is_ontology_expansion: bool = True,
+    is_silent_not_found_error: bool = True,
 ) -> Set[str]:
+    """Does ontology expansion on the given OMOP concept codes if `is_ontology_expansion` is True,
+        otherwise just returns the codes as given.
+
+    If `is_silent_not_found_error` is True, then this function will NOT raise an error
+        if a given OMOP concept ID is not found in the ontology.
+    """
+    if not isinstance(omop_concept_codes, list):
```

Comment:
This does not match the type signature. Maybe the type should be omop_concept_code: Union[str, List[str]] or something like that

Reply:
fixed!